solidity/test/bridge/Bridge.test.ts

```diff
@@ -1423,6 +1448,704 @@ describe("Bridge", () => {
+                        )
+
+                        context(
+                          "when redeemer has not made a sufficient allowance in Bank",
+                          () => {
+                            it("should revert", async () => {
+                              // TODO: Implementation.
+                            })
+                          }
+                        )
+                      })
+
+                      context("when wallet has insufficient funds", () => {
+                        it("should revert", async () => {
+                          // TODO: Implementation.
+                        })
+                      })
+                    }
+                  )
+
+                  context(
+                    "when there is a pending request for the given redemption key",
+                    () => {
+                      it("should revert", async () => {
+                        // TODO: Implementation.
+                      })
+                    }
+                  )
+                })
+
+                context("when amount is below the dust threshold", () => {
+                  it("should revert", async () => {
+                    // TODO: Implementation.
+                  })
+                })
+              }
+            )
+
+            context(
+              "when redeemer output script points to the wallet public key hash",
+              () => {
+                it("should revert", async () => {
+                  // TODO: Implementation. Make sure there is not possibility
+                  //       to pass the 20-byte wallet PKH under P2PKH, P2WPKH
+                  //       and P2SH.
+                })
+              }
+            )
+          })
+
+          context("when redeemer output script is not standard type", () => {
+            it("should revert", async () => {
+              // TODO: Implementation.
+            })
+          })
+        })
+
+        context("when main UTXO data are invalid", () => {
+          it("should revert", async () => {
+            // TODO: Implementation.
+          })
+        })
+      })
+
+      context("when there is no main UTXO for the given wallet", () => {
+        it("should revert", async () => {
+          // TODO: Implementation.
+        })
+      })
+    })
+
+    context("when wallet state is other than active", () => {
+      it("should revert", async () => {
+        // TODO: Implementation.
+      })
+    })
+  })
+
+  describe("submitRedemptionProof", () => {
+    context("when transaction proof is valid", () => {
+      context("when there is a main UTXO for the given wallet", () => {
+        context("when main UTXO data are valid", () => {
+          context("when there is only one input", () => {
+            context(
+              "when the single input points to the wallet's main UTXO",
+              () => {
+                context(
+                  "when wallet state is either Active or MovingFunds",
```

Comment:
This is testing only `Active` state, right? What is we set the name of the context to `when wallet state is Active` and add another context later, `when wallet state is MovingFunds` and just make sure the function works without repeating the same checks? Would be good to really cover that this function is callable both with `Active` and with `MovingFunds`

Reply:
I had a plan to just prepare a test data set with a `MovingFunds` wallet but we can make it in the way you proposed: https://github.com/keep-network/tbtc-v2/pull/128/commits/bd7b48fc10fcb38ee744a3df0676a2919130ab1c